Update logger methods

Mayisha opened the pull request:
## Changes proposed in this Pull Request:
In some places, the deprecated `WC_Stripe_Logger::log` method was still in use. In this PR, I have updated all the instances of `WC_Stripe_Logger::log` with the most suitable logger method, `info`, `error` etc. (I changed all the occurrences so it might have touched a few unused methods as well)

## Testing instructions
Code review.

## Note
I have updated the phpstan baseline in this PR, which increases the count of errors related to `WC_Subscriptions` class. The errors will be removed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4924

### Changelog entry

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

Mayisha committed:
update logger methods

Mayisha committed:
add changelog

Mayisha committed:
fix phpstan

Mayisha committed:
Merge branch 'develop' into tweak/use-correct-logger-method

Mayisha committed:
fix phpstan error

wjrosa commented on includes/abstracts/abstract-wc-stripe-payment-gateway.php:
```diff
@@ -1939,7 +1939,7 @@ public function setup_intent( $order, $prepared_source ) {
 
 		if ( is_wp_error( $setup_intent ) ) {
-			WC_Stripe_Logger::log( "Unable to create SetupIntent for Order #$order_id: " . print_r( $setup_intent, true ) );
+			WC_Stripe_Logger::error( "Unable to create SetupIntent for Order #$order_id: " . print_r( $setup_intent, true ) );
```
We can move the `$setup_intent` data to the `context` parameter to improve how the log is presented

wjrosa commented on includes/class-wc-stripe-helper.php:
```diff
@@ -1586,7 +1586,7 @@ public static function maybe_log_ip_issues( $ip_address ) {
 			}
 
-			WC_Stripe_Logger::log( 'Invalid IP address detected. Data: ' . wp_json_encode( $log_data ) );
+			WC_Stripe_Logger::warning( 'Invalid IP address detected. Data: ' . wp_json_encode( $log_data ) );
```
We can move the `$log_data` to the context parameter here as well.

wjrosa commented on includes/class-wc-stripe-intent-controller.php:
```diff
@@ -292,8 +292,8 @@ public function create_setup_intent() {
-				WC_Stripe_Logger::log( 'Failed create Setup Intent while saving a card.' );
-				WC_Stripe_Logger::log( "Response: $error_response_message" );
+				WC_Stripe_Logger::error( 'Failed create Setup Intent while saving a card.' );
+				WC_Stripe_Logger::error( "Response: $error_response_message" );
```
We can merge these two logs and move the response to the context parameter.

wjrosa commented on includes/connect/class-wc-stripe-connect.php:
```diff
@@ -512,7 +512,7 @@ public function refresh_connection() {
 				update_option( 'wc_stripe_' . $prefix . 'oauth_last_failed_at', time() );
 
-				WC_Stripe_Logger::log( 'OAuth connection refresh failed: ' . print_r( $response, true ) ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_print_r
+				WC_Stripe_Logger::error( 'OAuth connection refresh failed: ' . print_r( $response, true ) ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_print_r
```
We can move the response here to the context parameter as well

wjrosa reviewed (approved):
Thanks for this refactor! Left some suggestions

daledupreez commented on includes/abstracts/abstract-wc-stripe-payment-gateway-voucher.php:
```diff
@@ -282,7 +282,7 @@ public function process_payment( $order_id, $retry = true, $force_save_save = fa
 		} catch ( WC_Stripe_Exception $e ) {
 			wc_add_notice( $e->getLocalizedMessage(), 'error' );
-			WC_Stripe_Logger::log( 'Error: ' . $e->getMessage() );
+			WC_Stripe_Logger::error( 'Error: ' . $e->getMessage() );
```
Note that in some cases, `$e->getMessage()` returns the serialised response object, so we may want to consider pushing the message to the context. This may also make sense as something to tackle in a follow-up PR, as it's addressing a slightly different issue.
```suggestion
			WC_Stripe_Logger::error( 'Exception processing voucher payment', [ 'exception_message' => $e->getMessage() ] );
```

daledupreez commented on includes/abstracts/abstract-wc-stripe-payment-gateway.php:
```diff
@@ -1162,7 +1162,7 @@ public function update_fees( $order, $balance_transaction_id ) {
 			}
 		} else {
-			WC_Stripe_Logger::log( 'Unable to update fees/net meta for order: ' . $order->get_id() );
+			WC_Stripe_Logger::warning( 'Unable to update fees/net meta for order: ' . $order->get_id() );
```
Similar to my previous comment, is it worth adding the error details to the context?
```suggestion
			WC_Stripe_Logger::warning( 'Unable to update fees/net meta for order: ' . $order->get_id(), [ 'error' => $balance_transaction->error ] );
```

daledupreez commented on includes/abstracts/abstract-wc-stripe-payment-gateway.php:
```diff
@@ -1259,7 +1259,7 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
 			}
 		} catch ( WC_Stripe_Exception $e ) {
-			WC_Stripe_Logger::log( 'Error: ' . $e->getMessage() );
+			WC_Stripe_Logger::error( 'Error: ' . $e->getMessage() );
```
As above RE some exceptions including the full response object contents. We can consider improving the content in this PR or a follow-up.
```suggestion
			WC_Stripe_Logger::error( 'Exception processing refund', [ 'exception' => $e->getMessage() ] );
```

daledupreez commented on includes/abstracts/abstract-wc-stripe-payment-gateway.php:
```diff
@@ -1777,8 +1777,8 @@ private function get_intent( $intent_type, $intent_id ) {
-			WC_Stripe_Logger::log( "Failed to get Stripe intent $intent_type/$intent_id." );
-			WC_Stripe_Logger::log( "Response: $error_response_message" );
+			WC_Stripe_Logger::error( "Failed to get Stripe intent $intent_type/$intent_id." );
+			WC_Stripe_Logger::error( "Response: $error_response_message" );
```
I think we should log only one entry here, largely so we can avoid having duplicate stack traces.
```suggestion
			WC_Stripe_Logger::error( "Failed to get Stripe intent $intent_type/$intent_id.", [ 'response' => $response ] );
```

daledupreez commented on includes/class-wc-stripe-account.php:
```diff
@@ -403,7 +403,7 @@ public function is_webhook_enabled() {
 			return 'enabled' === $webhook_status;
 		} catch ( Exception $e ) {
-			WC_Stripe_Logger::log( 'Unable to determine webhook status: .;' . $e->getMessage() );
+			WC_Stripe_Logger::error( 'Unable to determine webhook status: .;' . $e->getMessage() );
```
This message looks like it could be cleaned up and possibly have the exception message shifted to the context.
```suggestion
			WC_Stripe_Logger::error( 'Unable to determine webhook status: ' . $e->getMessage() );
```

daledupreez commented on includes/deprecated/class-wc-stripe-apple-pay.php:
```diff
@@ -20,7 +20,7 @@ public function __construct() {
 
 		public static function instance() {
-			WC_Stripe_Logger::log( 'DEPRECATED! WC_Stripe_Apple_Pay class has been hard deprecated. Please remove any code that references this class or instance. This class will be removed by version 4.2' );
+			WC_Stripe_Logger::warning( 'DEPRECATED! WC_Stripe_Apple_Pay class has been hard deprecated. Please remove any code that references this class or instance. This class will be removed by version 4.2' );
```
Can we remove this class now? 4.2 was a _long_ time ago!

Mayisha replied:
Good call. 👍

daledupreez reviewed (approved):
The changes are looking good to me overall. Thanks for cleaning this up!

There are one or two obvious places to use a single log call, and there are a number of other cases where we may want a separate PR to push more details into the context, especially in cases where we're logging details from caught exceptions.

Mayisha committed:
Merge branch 'develop' into tweak/use-correct-logger-method

Mayisha committed:
update changelog statement

Mayisha committed:
move details from inline to context

Mayisha committed:
Merge branch 'develop' into tweak/use-correct-logger-method

Mayisha committed:
add context in more logs

Mayisha committed:
add context in remaining logs

Mayisha committed:
fix phpstan

Mayisha committed:
update baseline

Mayisha committed:
remaining items from review

Mayisha commented:
@daledupreez / @wjrosa could any of you give another 👀 into this PR? I have updated the log statements in some places and added the error as context in most of the places.

wjrosa reviewed (approved):
Thanks for addressing the suggestions! Looks good to me 👍
